src/core/ddsi/src/q_sockwaitset.c

```diff
@@ -60,7 +60,7 @@ struct entry {
 struct os_sockWaitset
 {
   int kqueue;
-  os_handle pipe[2];             /* pipe used for triggering */
+  os_socket pipe[2];             /* pipe used for triggering */
```

Comment:
Agreed. I've chosen os_socket because it was defined as such in the abstraction layer already. Replacing os_socket by os_handle everywhere is a simple search replace and shouldn't be too much work.

Reply:
The os_socket is the proper abstraction in most places except for the os_sockWaitset. One of the first things I'm going to do after this PR is merged, is push that down to the abstraction layer. Is it ok with you if I take care of it then?

Reply:
Yep, that's fine. Some as with the comment below on another instance of the same phenomenon.